client/src/main/java/com/vesoft/nebula/meta/client/MetaClientImpl.java

```diff
@@ -6,21 +6,297 @@
+            int port = address.getPort();
+            if (!InetAddresses.isInetAddress(host) || (port <= 0 || port >= 65535)) {
+                throw new IllegalArgumentException(String.format("%s:%d is not a valid address",
+                        host, port));
+            }
+        });
+        this.parts = Maps.newHashMap();
+        this.tagItems = Maps.newHashMap();
+        this.edgeItems = Maps.newHashMap();
+        this.addresses = addresses;
+        this.timeout = timeout;
+        this.connectionRetry = connectionRetry;
 
+        this.init();
     }
 
+    public MetaClientImpl(String host, int port) {
+        this(Lists.newArrayList(HostAndPort.fromParts(host, port)), DEFAULT_TIMEOUT_MS, DEFAULT_CONNECTION_RETRY_SIZE);
+    }
+
+    public MetaClientImpl(List<HostAndPort> addresses) {
+        this(addresses, DEFAULT_TIMEOUT_MS, DEFAULT_CONNECTION_RETRY_SIZE);
+    }
+
+    /**
+     * Get a list of host addresses by a particular space Id
+     *
+     * @param spaceId
+     * @param partId
+     * @return
+     */
     @Override
+    public List<HostAddr> getPart(int spaceId, int partId) {
+        if (!this.parts.containsKey(spaceId)) {
+            getParts(spaceId);
+        }
+        Map<Integer, List<HostAddr>> map = parts.get(spaceId);
+        if (map.isEmpty()) return null;
+        List<HostAddr> addrs = map.get(partId);
+        return addrs.isEmpty() ? null : addrs;
+    }
+
+    @Override
+    public List<HostAddr> getPart(String spaceName, int partId) {
+        if(!spaceNames.containsKey(spaceName)) {
+            LOGGER.error("There is no space named: %s", spaceName);
+            return null;
+        }
+        return getPart(spaceNames.get(spaceName).getId().getSpace_id(), partId);
+    }
+
+    /**
+     * Get a tag Id by a particular space Id and tag name
+     *
+     * @param spaceId
+     * @param tagName
+     * @return
+     */
+    @Override
+    public Integer getTagId(int spaceId, String tagName) {
+        if (!this.tagItems.containsKey(spaceId)) {
+            getTagItems(spaceId);
+        }
+        Map<String, TagItem> map = tagItems.get(spaceId);
+        if (map.isEmpty()) return null;
+        TagItem tag = map.get(tagName);
+        return tag == null ? null : tag.getTag_id();
+    }
+
+    /**
+     * Get a edge type by a particular space Id and edge name
+     *
+     * @param spaceId
+     * @param edgeName
+     * @return
+     */
+    @Override
+    public Integer getEdgeType(int spaceId, String edgeName) {
+        if (!this.edgeItems.containsKey(spaceId)) {
+            getEdgeTypes(spaceId);
+        }
+        Map<String, EdgeItem> map = edgeItems.get(spaceId);
+        if (map.isEmpty()) return null;
+        EdgeItem edge = map.get(edgeName);
+        return edge == null ? null : edge.getEdge_type();
+    }
+
+    private void init() {
+        connect();
+        listSpaces();
+        for (IdName spcace : spaces) {
+            spaceNames.put(spcace.getName(), spcace);
```

Comment:
spaceNames is space name to space ID would be better ?  and maybe could rename it .